vega_sim/scenario/ideal_market_maker/agents.py

```diff
@@ -572,3 +589,277 @@ def initialise(self, vega: VegaServiceNull):
+        self.current_step = 0
+        self.sim_length = len(price_process)
+        self.tag = tag
+        self.market_name = f"ETH:USD_{self.tag}" if market_name is None else market_name
+        self.asset_name = f"tDAI_{self.tag}" if asset_name is None else asset_name
+
+    def initialise(self, vega: VegaServiceNull):
+        # Initialise wallet
+        super().initialise(vega=vega)
+
+        # Get market id
+        self.market_id = [
+            m.id
+            for m in self.vega.all_markets()
+            if m.tradable_instrument.instrument.name == self.market_name
+        ][0]
+
+        # Get asset id
+        tDAI_id = self.vega.find_asset_id(symbol=self.asset_name)
+        # Top up asset
+        self.vega.mint(
+            self.wallet_name,
+            asset=tDAI_id,
+            amount=self.initial_asset_mint,
+        )
+        self.vega.wait_for_total_catchup()
+
+    def step(self, vega_state: VegaState):
+        position = self.vega.positions_by_market(
+            wallet_name=self.wallet_name, market_id=self.market_id
+        )
+        current_position = int(position[0].open_volume) if position else 0
+        trade_side = (
+            vega_protos.vega.Side.SIDE_BUY
+            if current_position < 0
+            else vega_protos.vega.Side.SIDE_SELL
+        )
+        if current_position:
+            self.vega.submit_market_order(
+                trading_wallet=self.wallet_name,
+                market_id=self.market_id,
+                side=trade_side,
+                volume=current_position,
+                wait=True,
+                fill_or_kill=False,
+            )
+
+        Order_book = []
+        for _, orders in (
+            self.vega.order_status_from_feed(live_only=True)
+            .get(self.market_id, {})
+            .items()
+        ):
+            Order_book += list(orders.values())
+
+        price = self.price_process[self.current_step]
+        next_price = (
+            self.price_process[self.current_step + 1]
+            if self.sim_length > self.current_step
+            else price
+        )
+
+        trade_side = (
+            vega_protos.vega.Side.SIDE_BUY
+            if price < next_price
+            else vega_protos.vega.Side.SIDE_SELL
+        )
+        volume = 0
+        if price < next_price:
+            for order in Order_book:
+                if order.side != trade_side and order.price < next_price:
+                    volume += order.remaining
+        else:
+            for order in Order_book:
+                if order.side != trade_side and order.price > next_price:
+                    volume += order.remaining
+
+        if volume:
+            self.vega.submit_market_order(
+                trading_wallet=self.wallet_name,
+                market_id=self.market_id,
+                side=trade_side,
+                volume=volume,
+                wait=True,
+                fill_or_kill=False,
+            )
```

Comment:
Could we move this to the `common` agents repository? Seems a useful one in general. 

Would also be cool to add a `proportion_taken` (or better name) config parameter to it which could multiply with the `volume` here so that it doesn't have to take *all* of the 'informed' price advantage.

Reply:
The idea of `proportion_take` is cool, will add it.